projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx

```diff
@@ -58,20 +56,21 @@ export default function ProductInterstitial( {
 	const navigateToMyJetpackOverviewPage = useMyJetpackNavigate( '/' );
 
 	const clickHandler = useCallback(
-		checkout => {
-			const activateOrCheckout = () => ( bundle ? Promise.resolve() : activate() );
+		( checkout, selectedProductDetail ) => {
+			const activateOrCheckout = () =>
+				selectedProductDetail?.isBundle ? Promise.resolve() : activate();
```

Comment:
In this `clickHandler()` function, `bundle` is referencing the bundle that the product is upgradable by, i.e.
```javascript
/**
 * ScanInterstitial component
 *
 * @returns {object} ScanInterstitial react component.
 */
export function ScanInterstitial() {
	return <ProductInterstitial slug="scan" installsPlugin={ true } bundle="security" />;
}
```
Whereas `selectedProductDetail?.isBundle` indicates whether the product we're showing is a bundle itself. Is this change intentional? I will double-check for side effects.

Reply:
Sorry! I missed this comment. The idea behind this is to have selectedProductDetail to be an object representing what the user selected on the page, in our case it can either be a bundle ('security') or just the Protect plugin (which isn't a bundle), that way we can have different behaviors depending on what was clicked.

The way the page worked before is that it expected the other side of the bundle to also be a product that would redirect to the checkout with it's own slug. We have the unique case of being a free product that doesn't need checkout that is upgradeable with a bundle, that's why we need this different handling.

Reply:
I understand now, I hadn't realized we were exclusively using the existing "bundle" property on `ProductInterstitial` only for the security bundle cases in the first place, so this change makes sense! :+1: :+1: